src/main/java/ProductPackage/ShopService.java

```diff
@@ -0,0 +1,46 @@
+package ProductPackage;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class ShopService {
+
+    private ProductRepo productRepo;
+    private OrderRepo orderRepo = new OrderRepo();
+
+    public ShopService(ProductRepo productRepo) {
+        this.productRepo = productRepo;
+    }
+
+    public Product getProductById(String id) {
+        Optional<Product> currentProduct = productRepo.getProductById(id);
+        return currentProduct.orElseThrow(() -> new RuntimeException("Product does not exist"));
+    }
+
+    public List<Product> listProducts() {
+        return productRepo.list();
+    }
+
+    public void addOrder(List<String> idList) {
+        List<Product> productList = new ArrayList<>();
+        for(String id : idList) {
+            Optional<Product> product = productRepo.getProductById(id);
```

Comment:
Das innere der for-Schleife kannst du auch ein bisschen kompakter so schreiben:
```java
productList.add(productRepo.getProductById(id)
        .orElseThrow(new RuntimeException("Product does not exist")));
```
Nächste Woche schauen wir uns dann an wie man dafür auch Streams hätte nutzen können.